src/libstore/derivations.cc

```diff
@@ -267,6 +267,28 @@ static DerivationOutput parseDerivationOutput(const Store & store, std::istrings
 }
 
 
+static DerivedPathMap<StringSet>::ChildNode parseDerivedPathMapNode(const Store & store, std::istringstream & str)
+{
+    DerivedPathMap<StringSet>::ChildNode node;
+    node.value = parseStrings(str, false);
+    char c = str.get();
+    switch (c) {
+    case ')':
+        return node;
+    case ',':
+        expect(str, "[");
```

Comment:
This would be the right thing to do in a vacuum, but wouldn't we get a slightly better error if we repurposed the `^` character here? Then old Nix would report something like `output not found 'foo^bar'`, which is a slightly better hint than something like `couldn't parse deadbeef.drv at char 123`, I think.

(And *maybe* we get an even better error thanks to `requiredFeatures` if we do that. May well be wishful thinking though; seems like a later stage...)

Reply:
The new error should cause it to fail with

> `expected "\""`

which is not ideal, but better than it "getting off" and trying to parse the next bit after `inputDrvs` as `inputDrvs`.

Reply:
Wouldn't a format like this give an even better error, or is that not feasible?

```aterm
[
  "cat",
  "dog",
  "cat^kitten",
  "goose^gosling"
]
```

Reply:
ie avoiding the `outputs` × `deeper`, but rather make use of the unused space within the individual output syntax.

Reply:
(recapping live discussion for others) I made the format self-versioning to try to make the error better while keeping the deduplication we currently have that would go away with the flat list of `^`